view/Requirements.php

```diff
@@ -1332,6 +1349,38 @@ public function themedCSS($name, $module = null, $media = null) {
 	}
 
 	/**
+	 * Registers the given themeable javascript as required.
+	 *
+	 * A javascript file in the current theme path name 'themename/javascript/$name.js' is first searched for,
+	 * and it that doesn't exist and the module parameter is set then a javascript file with that name in
+	 * the module is used.
+	 *
+	 * @param string $name   The name of the file - eg '/js/File.js' would have the name 'File'
+	 * @param string $module The module to fall back to if the javascript file does not exist in the
+	 *                       current theme.
+	 * @param string $type  Comma-separated list of types to use in the script tag
+	 *                       (e.g. 'text/javascript,text/ecmascript')
+	 */
+	public function themedScript($name, $module = null, $type = null) {
+		$theme = SSViewer::get_theme_folder();
+		$project = project();
+		$absbase = BASE_PATH . DIRECTORY_SEPARATOR;
+		$abstheme = $absbase . $theme;
+		$absproject = $absbase . $project;
+		$js = "/javascript/$name.js";
+
+		if(file_exists($absproject . $js)) {
+			$this->javascript($project . $js);
+		} elseif($module && file_exists($abstheme . '_' . $module.$js)) {
+			$this->javascript($theme . '_' . $module . $js);
+		} elseif(file_exists($abstheme . $js)) {
+			$this->javascript($theme . $js);
+		} elseif($module) {
```

Comment:
That is a great addition! I will make the changes in both themedCSS and themedJavascript and will throw the exception only if the site is in dev mode.